src/core_plugins/timelion/public/directives/expression_directive.js

```diff
@@ -72,22 +80,25 @@ app.directive('timelionExpressionInput', function ($compile, $http, $timeout) {
         });
       }
 
+      function setExpression(expression, caretPosition) {
+        input.val(expression);
+        input[0].selectionStart = input[0].selectionEnd = caretPosition;
+      }
+
       function completeExpression(suggestionIndex) {
-        if ($scope.functionSuggestions.isEmpty()) {
+        if (scope.functionSuggestions.isEmpty()) {
           return;
         }
 
-        const functionName = `${$scope.functionSuggestions.list[suggestionIndex].name}()`;
-        const expression = $scope.sheet;
+        const functionName = `${scope.functionSuggestions.list[suggestionIndex].name}()`;
+        const expression = scope.sheet;
         const { min, max } = caretLocation;
 
         const newExpression = insertAtLocation(functionName, expression, min, max);
-        input.val(newExpression);
-
         const newCaretPosition = min + functionName.length - 1;
-        input[0].selectionStart = input[0].selectionEnd = newCaretPosition;
+        setExpression(newExpression, newCaretPosition);
 
-        $scope.functionSuggestions.reset();
+        scope.functionSuggestions.reset();
```

Comment:
Beyond this point I pretty much just renamed $scope to scope, since the $ is used to indicate a service injected via Angular's DI, but the scope here is just a regular function parameter.